Algorithms/Search/Substring/RabinKarp.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+
+namespace Algorithms.Search.Substring
+{
+    /// <summary>
+    /// The idea: You calculate the hash for the pattern <c>p</c> and the hash values for all the prefixes of the text <c>t</c>.
+    /// Now, you can compare a substring in constant time using the calculated hashes.
+    /// time complexity: O(p + t),
+    /// space complexity: O(t),
+    /// where   t - text length
+    ///         p - pattern length.
+    /// </summary>
+    public class RabinKarp
```

Comment:
Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6278031)